internal/netclassify/classify_test.go

```diff
@@ -97,6 +97,77 @@ func TestClassifyURL(t *testing.T) {
 			wantErr: ErrUnclassified,
 		},
 
+		// NPM test cases
+		{
+			name: "npm_download_simple",
+			url:  "https://registry.npmjs.org/express/-/express-4.17.1.tgz",
+			want: "pkg:npm/express@4.17.1",
+		},
+		{
+			name: "npm_download_scoped",
+			url:  "https://registry.npmjs.org/@invisionag/eslint-config-ivx/-/eslint-config-ivx-0.0.2.tgz",
+			want: "pkg:npm/@invisionag/eslint-config-ivx@0.0.2",
+		},
+		{
+			name: "npm_yarn_download_simple",
+			url:  "https://registry.yarnpkg.com/express/-/express-4.17.1.tgz",
+			want: "pkg:npm/express@4.17.1",
+		},
+		{
+			name:    "npm_api_scoped",
+			url:     "https://registry.npmjs.org/@esbuild/freebsd-arm64/0.21.5",
+			wantErr: ErrSkipped,
+		},
+		{
+			name:    "npm_api_simple",
+			url:     "https://registry.npmjs.org/express/4.17.1",
+			wantErr: ErrSkipped,
+		},
+		{
+			name:    "npm_yarn_api_simple",
+			url:     "https://registry.yarnpkg.com/express/4.17.1",
+			wantErr: ErrSkipped,
+		},
+
+		// Maven test cases
+		{
+			name: "maven_central_artifact",
+			url:  "https://repo1.maven.org/maven2/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar",
+			want: "pkg:maven/org.apache.commons/commons-lang3@3.12.0",
+		},
+		{
+			name: "maven_with_classifier",
+			url:  "https://repo1.maven.org/maven2/org/apache/spark/spark-core_2.12/3.1.2/spark-core_2.12-3.1.2-tests.jar",
+			want: "pkg:maven/org.apache.spark/spark-core_2.12@3.1.2",
+		},
+		{
+			name: "maven_gradle_plugin",
```

Comment:
gradle_plugin_repo_artifact

Reply:
given the regex naming, i like this provided we keep the maven_ prefix